src/Templates/InterfacesTemplate.cshtml

```diff
@@ -0,0 +1,81 @@
+@using AutoRest.Core.Utilities
+@using AutoRest.Go
+@using AutoRest.Go.Model
+@using AutoRest.Go.Templates
+@using System;
+@using System.Collections.Generic;
+@using System.Linq;
+
+@inherits AutoRest.Core.Template<AutoRest.Go.Model.CodeModelGo>
+
+@{
+    var content = new Dictionary<string, IEnumerable<MethodGo>>();
+    var imports = new HashSet<string> { PrimaryTypeGo.GetImportLine(package: "context") };
+
+    if (Model.ClientMethods.Any())
+    {
+        content.Add(Model.BaseClient, Model.ClientMethods.OrderBy(m => m.Name.Value));
+    }
+
+    var methodGroups = Model.MethodGroups.Where(mg => !string.IsNullOrEmpty(mg.Name));
+    if (methodGroups.Any())
+    {
+        foreach (var methodGroup in methodGroups)
+        {
+            content.Add(methodGroup.ClientName, methodGroup.Methods.Cast<MethodGo>().OrderBy(m => m.Name.Value));
+        }
+    }
+
+    var addArImport = false;
+    foreach (var methods in content.Values)
+    {
+        foreach (var method in methods)
+        {
+            foreach (var p in method.Parameters)
+            {
+                p.ModelType.AddImports(imports);
+            }
+            if (method.HasReturnValue())
+            {
+                method.ReturnType.Body.AddImports(imports);
+            }
+            else
+            {
+                addArImport = true;
+            }
+        }
+    }
+
+    var arImport = PrimaryTypeGo.GetImportLine(package: "github.com/Azure/go-autorest/autorest");
```

Comment:
so this is needed only if there are methods without a return value ? is that the only case when that import is necessary ? in that case can we extract a isAutorestImportNeeded method or something like that